### PR TITLE
Update MaterialParams to store different materials with same name

### PR DESCRIPTION
**Pull request by stognini**
This PR fixes the issue regarding geometries (such as `cms2018.gdml`) that define different materials with the same string name. Celeritas skipped the import of these cases, since duplicate `pair<"name", MaterialId>` are not allowed. 

The updated `MaterialParams` creates a unique name identifier in these cases by appending the `MaterialId` at the end of the name.

### CODE REVIEW

File: src/physics/material/MaterialParams.cc [hunk 1]
```diff
@@ -157,13 +157,27 @@ void MaterialParams::append_material_def(const MaterialInput& inp,
...
-        // Insertion failed, so material name is a duplicate
-        CELER_LOG(warning)
-            << "Material " << inp.name << " already exists with id "
-            << iter_inserted.second << ". This new id ("
-            << host_data->materials.size() << ") will not be available.";
+        // Insertion failed due to duplicate material name
+        // Create unique material name by concatenating its name and MaterialId
+        mat_name
+            = inp.name + "_"
+              + std::to_string(MaterialId(host_data->materials.size()).get());
```
Comment (sethrj) [inline src/physics/material/MaterialParams.cc:169]:
```suggestion
              + std::to_string(host_data->materials.size());
```

**Review by sethrj — CHANGES_REQUESTED** [review]
Looks great, a better stopgap measure than what's in place. It's still a little fragile since (e.g.) reordering the materials will give very different names but at least now we can access all the materials.

I have one minor simplification to suggest.

**Comment by stognini** [comment]
> It's still a little fragile since (e.g.) reordering the materials will give very different names but at least now we can access all the materials

At this point it is unlikely that we can reorder materials (unless we add another layer of indirection), since the material id also represents the position of said material xs data in the imported physics tables.

**Comment by sethrj** [comment]
> > It's still a little fragile since (e.g.) reordering the materials will give very different names but at least now we can access all the materials
> 
> At this point it is unlikely that we can reorder materials (unless we add another layer of indirection), since the material id also represents the position of said material xs data in the imported physics tables.

Agreed, we should def preserve the order of material IDs; I was suggesting the possibility of an advanced but robust scheme where we take the material definitions, and do a sorted map of names/densities/states/components to ID, and *then* add suffixes based on which comes first. This would give us reproducibility while keeping the IDs correct. But it would also be  a lot of work -- so  let's wait until someone complains about the matid suffixes ;)

**Review by sethrj — APPROVED** [review]

**Pull request merged**